File: src/WebApp/Application/Models/SearchData.cs
using Application.Models.Genres;
using Application.Models.Playlists;
using Application.Models.Songs;
using Application.Models.Users;

namespace Application.Models;
public class SearchData
{
    public string BestFitType { get; set; } = "";
    public object? BestFitItem { get; set; } = null;
    public PagedList<SongResponse> Songs { get; set; } = null!;
    public PagedList<PlaylistResponse> Playlists { get; set; } = null!;
    public PagedList<GenreResponse> Genres { get; set; } = null!;
    public PagedList<UserResponse> Users { get; set; } = null!;
    public bool HasItems { get; set; } = false;
}

File: src/WebClient/src/app/core/models/searchData.ts
import { GenreListResponse } from "./genre/genreListResponse";
import { PlaylistListResponse } from "./playlist/playlistListResponse";
import { SongListResponse } from "./song/songListResponse";
import { UserListResponse } from "./user/userListResponse";

export interface SearchData {
  bestFitType: string;
  bestFitItem: unknown;
  songs: SongListResponse;
  playlists: PlaylistListResponse;
  genres: GenreListResponse;
  users: UserListResponse;
  hasItems: boolean;
}